src/pages/home/home.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import styles from "./home.module.scss";
+import GamesBlock from "@/components/blocks/gameCardsBlock";
+import CategoriesBlock from "@/components/blocks/categoriesBlock";
+import Search from "@/elements/search/search";
+import { ProductInfo } from "@/productInfos";
+import { selectProductInfos } from "@/api/clientRequests/getProductInfos";
+
+function debounce<A = unknown, R = void>(fn: (args: A) => R, ms: number): (args: A) => Promise<R> {
+  let timer: NodeJS.Timeout;
+
+  const debouncedFunc = (args: A): Promise<R> =>
+    new Promise((resolve) => {
+      if (timer) {
+        clearTimeout(timer);
+      }
+
+      timer = setTimeout(() => {
+        resolve(fn(args));
+      }, ms);
+    });
+
+  return debouncedFunc;
+}
```

Comment:
this is a universal function/ replace it to utils folder

Reply:
/utils folder created and debounce function replaced in it